SpatialGDK/Source/SpatialGDK/Private/Interop/SkeletonEntities.cpp

```diff
@@ -0,0 +1,479 @@
+		   Manifest.PopulatedEntities.Num(), EntityId);
+
+	Manifests.Add(EntityId, Manifest);
+}
+
+FSkeletonEntityPopulator::FSkeletonEntityPopulator(USpatialNetDriver& InNetDriver)
+	: NetDriver(&InNetDriver)
+	, SubView(&InNetDriver.Connection->GetCoordinator().CreateSubView(SpatialConstants::FLESHOUT_REQUIRED_TAG_COMPONENT_ID,
+																	  FSubView::NoFilter, FSubView::NoDispatcherCallbacks))
+{
+	Stage = EStage::ReceivingManifest;
+}
+
+void FSkeletonEntityPopulator::ConsiderEntityPopulation(const Worker_EntityId EntityId, const EntityViewElement& Element)
+{
+	const ComponentData* UnrealMetadataComponent = Element.Components.FindByPredicate(ComponentIdEquality{ UnrealMetadata::ComponentId });
+
+	if (!ensure(UnrealMetadataComponent != nullptr))
+	{
+		return;
+	}
+
+	if (!ensure(Manifest->EntitiesToPopulate.Contains(EntityId)))
+	{
+		return;
+	}
+
+	const UnrealMetadata ActorMetadata(UnrealMetadataComponent->GetUnderlying());
+
+	if (ActorMetadata.bNetStartup.IsSet() && ActorMetadata.bNetStartup.GetValue() && ActorMetadata.StablyNamedRef.IsSet())
+	{
+		const FUnrealObjectRef& ActorRef = ActorMetadata.StablyNamedRef.GetValue();
+		bool bUnresolved = false;
+		UObject* Object = FUnrealObjectRef::ToObjectPtr(ActorRef, NetDriver->PackageMap, bUnresolved);
+		if (!ensure(!bUnresolved))
+		{
+			return;
+		}
+
+		AActor* StartupActor = Cast<AActor>(Object);
+		PopulateEntity(EntityId, *StartupActor);
+
+		Manifest->PopulatedEntities.Emplace(EntityId);
+	}
+
+	EntitiesFleshedOut.Emplace(EntityId);
+}
+
+void FSkeletonEntityPopulator::Advance()
+{
+	if (Stage == EStage::ReceivingManifest)
+	{
+		for (const auto& Kvp : GetCoordinator().GetView())
+		{
+			const Worker_EntityId EntityId = Kvp.Key;
+			const ComponentData* ManifestPtr = GetCoordinator().GetComponent(EntityId, FSkeletonEntityManifest::ComponentId);
+			if (ManifestPtr == nullptr)
+			{
+				continue;
+			}
+			if (!Kvp.Value.Authority.Contains(SpatialConstants::SKELETON_ENTITY_MANIFEST_AUTH_COMPONENT_SET_ID))
+			{
+				continue;
+			}
+			check(!ManifestEntityId);
+			ManifestEntityId = EntityId;
+			Manifest = FSkeletonEntityManifest(*ManifestPtr);
+		}
+		if (Manifest)
+		{
+			Stage = EStage::PopulatingEntities;
+		}
+	}
+
+	if (Stage == EStage::PopulatingEntities)
+	{
+		if (!bStartedPopulatingEntities)
+		{
+			bStartedPopulatingEntities = true;
+
+			for (const Worker_EntityId_Key CompleteEntities : SubView->GetCompleteEntities())
+			{
+				ConsiderEntityPopulation(CompleteEntities, NetDriver->Connection->GetCoordinator().GetView()[CompleteEntities]);
+			}
+		}
+		else
+		{
+			for (const EntityDelta& Delta : SubView->GetViewDelta().EntityDeltas)
+			{
+				if (Delta.Type == EntityDelta::ADD)
+				{
+					const Worker_EntityId EntityId = Delta.EntityId;
+					const EntityViewElement& Element = SubView->GetView()[Delta.EntityId];
+
+					ConsiderEntityPopulation(EntityId, Element);
+				}
+			}
+		}
+
+		if (OnManifestUpdated)
```

Comment:
Maybe this is "short-circuiting"?

Reply:
Another (possibly easier) hack we can do for this is to have it be a completeness thing. If you send a load of updates or add components etc then next tick it could appear as a completeness thing.

Reply:
Actually reimplemented this as completeness - we do need to call RefreshCompleteness but that's the lesser evil I guess.